front/src/app/todo-list/todo-list.component.ts

```diff
@@ -54,6 +62,33 @@ export class TodoListComponent implements OnInit, OnDestroy {
     this.destroy$.complete();
   }
 
+  displayInput(): void {
+    this.closeInput = false;
+    this.changeDetectorRef.detectChanges()
+    this.grpInp?.nativeElement.focus();
+  }
+
+  hideInput(): void {
+    this.closeInput = true;
+  }
+
+  createNewTskGroup(name: string): void {
+    this.load = true
+    if (name != '') {
+      this.apiService.createTaskGroup(name)
+        .pipe(takeUntil(this.destroy$))
+        .subscribe(data => {
+          let list = data.list.id;
+          this.changeDetectorRef.markForCheck();
+          this.loadLists();
+          this.router.navigate([], {queryParams: {list}})
+        })
+    }
+    this.closeInput = true;
+    this.load = false;
+    this.inputText = '';

```

Comment:
add delay to network request via corresponding rxjs operator and heck if you show spinner at all